### ISSUE TITLE
Why some modules are not being exported in index.ts?

### ISSUE THREAD

**Issue by JadenKim-dev** [woowabros/nestjs-library-config#414]
Hi, there.

First of all, thanks for all your great works.
This library resolved many inconveniences I experienced while using nest/config.

I wondered why source provider modules like `DotenvSourceProvider` are not exported in index.ts. 
It seems possible to specify the sourceProvider as an option in the ConfigModule.forFeature, but when I want to use DotenvSourceProvider, I can't import it.

I want to use as below.

```ts
// Module '"@nestjs-library/config"' has no exported member 'DotenvSourceProvider'.
import { ConfigModule, DotenvSourceProvider } from '@nestjs-library/config';

@Module({
    imports: [
        TypeOrmModule.forRootAsync({
            imports: [
                ConfigModule.forFeature(TypeOrmConfigService, {
                    sourceProvider: new DotenvSourceProvider(),
                }),
            ],
            useFactory: (typeORMConfigService: TypeOrmConfigService) => typeORMConfigService,
            inject: [TypeOrmConfigService],
        }),
    ],
    controllers: [AppController],
    providers: [AppService],
})
export class AppModule {}
```

Current `index.ts` is like below.
It exports only ConfigModule, AbstractConfigService, OptionalBoolean, ConfigScanner

```ts
export * from './lib/config.module';
export * from './lib/abstract/abstract-config.service';

export * from './lib/type-transformer';
export * from './lib/config-scanner';
```

**Comment by jiho-kr** [comment]
Hi, @JadenKim-dev 

As far as I know, there's no particular reason for this, it's simply missing.

The most likely reason is that my team doesn't use 'DotenvSourceProvider' and I didn't know about it. 😁 

And as a background on providers, 
they expect you to create and utilize your own, not have it provided by config library.

A DotenvSourceProvider was created as a guide for creating providers.

**Comment by JadenKim-dev** [comment]
Thank you for your kind explanation, @jiho-kr! 
Considering your points, it would be beneficial to also export the abstract class `AbstractConfigSourceProvider` to make it easier for users to implement their own provider. 
I will proceed with this task and create a PR afterward! Thanks.

**Comment by Pigrabbit** [comment]
It has been resolved in #415.

**Comment by Pigrabbit** [comment]
It has been resolved in https://github.com/woowabros/nestjs-library-config/pull/415.
